napari/layers/_vectors_layer/model.py

```diff
@@ -378,21 +422,41 @@ def _set_view_slice(self, indices):
         indices : sequence of int or slice
             Indices to slice with.
         """
-        
-        in_slice_vectors = self.vectors
 
-        # Display vectors if there are any in this slice
-        if len(in_slice_vectors) > 0:
-            data = np.array(in_slice_vectors) + 0.5
+        vertices = self._mesh_vertices
+        faces = self._mesh_triangles
+
+        if len(faces) == 0:
+            self._node.set_data(vertices=None, faces=None)
         else:
-            # if no vectors in this slice send dummy data
-            data = np.empty((0, 2))
-
-        self._node.set_data(
-            data[::-1],
-            width=self.width,
-            color=self.color,
-            connect='segments')
+            self._node.set_data(vertices=vertices, faces=faces,
+                                color=self.color)
+
         self._need_visual_update = True
         self._update()
 
+
+def segment_normal(a, b):
```

Comment:
This should go into some `vector_utils` module, or be made private, or both.

Reply:
moved